tests/test_resolver.py

```diff
@@ -109,35 +116,49 @@ def test_txt(self):
             ['a', 'b', 'c', 'd'])
 
     def test_csv_expr(self):
-        out = io.BytesIO()
-        writer = csv.writer(out)
-        writer.writerows([range(5) for r in range(5)])
-        values = self.get_values_from(
-            {'url': 'sun.csv', 'expr': '[*][2]'}, out.getvalue())
+        with open('test.csv', 'w') as out:
+            writer = csv.writer(out)
+            writer.writerows([range(5) for r in range(5)])
+        with open('test.csv', 'r') as out:
+            values = self.get_values_from(
+                {'url': 'sun.csv', 'expr': '[*][2]'},
+                out.read(),
+            )
+        os.remove('test.csv')
```

Comment:
I had it there, but `tearDown()` runs for every test and not all of them create that file.